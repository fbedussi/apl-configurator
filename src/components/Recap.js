import React from 'react';

function getAnswerLabel(node, answers, i) {
    var selectedAnswerKey = answers[i];
    var selectedAnswerArr = node.answers.filter(answer => {
        var answerKey = Object.keys(answer)[0];
        return answerKey === selectedAnswerKey
    });
    var selecterAnswerLAbel = selectedAnswerArr[0][selectedAnswerKey];
    return selecterAnswerLAbel
}

const Recap = ({ show, breadcrumbs, answers }) => {
    if (!show) {
        return null;
    }

    return <div className="recap">
        <h2 className="recap-title">Le tue risposte</h2>
        <table className="recap-table">
            <thead>
                <tr>
                    <th>Domanda</th>
                    <th>Risposta</th>
                </tr>
            </thead>
            <tbody>
                {breadcrumbs.map((node, i) => <tr>
                    <td>
                        {node.text}
                    </td>
                    <td>
                        {getAnswerLabel(node, answers, i)}
                    </td>
                </tr>)}
            </tbody>
        </table>
    </div>
}

export default Recap;