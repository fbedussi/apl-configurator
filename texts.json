{
    "it": {
        "id": 1,
        "type": "question",
        "text": "Nel sito di installazione è possibile collegare ol sistema APL alla rete elettrica?",
        "answers": [
            {
                "a": "no"
            },
            {
                "b": "sì"
            }
        ],
        "a": {
            "id": 2,
            "type": "answer",
            "title": "Apl solar",
            "text": "Questa soluzione è valida solo se nel sito di installazione non è presente illuminazione pubblica in quanto le potenze sviluppate non sono sufficienti a creare un contrasto adeguato se non in zone buie"
        },
        "b": {
            "id": 3,
            "type": "question",
            "text": "Il sistema deve essere dotato di dispositivo per l'interazione con il pedone?",
            "answers": [
                {
                    "a": "no"
                },
                {
                    "b": "sì"
                }
            ],
            "a": {
                "id": 4,
                "type": "question",
                "text": "Nel luogo di installazione è già presente illuminazione stradale oppure altra illuminazione artificiale? (Vetrine, insegne luminose, ecc.)?",
                "answers": [
                    {
                        "a": "no"
                    },
                    {
                        "b": "sì"
                    }
                ],
                "a": {
                    "id": 5,
                    "type": "question",
                    "text": "La distanza tra i 2 pali è maggiore di 9 metri?",
                    "answers": [
                        {
                            "a": "no"
                        },
                        {
                            "b": "sì"
                        }
                    ],
                    "a": {
                        "id": 6,
                        "type": "question",
                        "text": "Ci sono ostacoli dove verranno installati i pali che potrebbero ridurre la visibilità dei cartelli di segnalazione laterali (alberi, tabelle segnaletiche o pubblicitarie, ecc)?",
                        "answers": [
                            {
                                "a": "no"
                            },
                            {
                                "b": "sì"
                            }
                        ],
                        "a": {
                            "id": 7,
                            "type": "answer",
                            "title": "Apl classic Stratos N soluzione 1",
                            "text": "Testo"
                        },
                        "b": {
                            "id": 8,
                            "type": "answer",
                            "title": "Apl classic Stratos N soluzione 2",
                            "text": "Testo"
                        }
                    },
                    "b": {
                        "id": 9,
                        "type": "question",
                        "text": "Ci sono ostacoli dove verranno installati i pali che potrebbero ridurre la visibilità dei cartelli di segnalazione laterali (alberi, tabelle segnaletiche o pubblicitarie, ecc)?",
                        "answers": [
                            {
                                "a": "no"
                            },
                            {
                                "b": "sì"
                            }
                        ],
                        "a": {
                            "id": 10,
                            "type": "answer",
                            "title": "Apl classic Stratos P soluzione 1",
                            "text": "Testo"
                        },
                        "b": {
                            "id": 11,
                            "type": "answer",
                            "title": "Apl classic Stratos P soluzione 2",
                            "text": "Testo"
                        }
                    }
                },
                "b": {
                    "id": 12,
                    "type": "question",
                    "text": "Ci sono ostacoli dove verranno installati i pali che potrebbero ridurre la visibilità dei cartelli di segnalazione laterali (alberi, tabelle segnaletiche o pubblicitarie, ecc)?",
                    "answers": [
                        {
                            "a": "no"
                        },
                        {
                            "b": "sì"
                        }
                    ],
                    "a": {
                        "id": 13,
                        "type": "answer",
                        "title": "Apl classic Stratos P soluzione 1",
                        "text": "Testo"
                    },
                    "b": {
                        "id": 14,
                        "type": "answer",
                        "title": "Apl classic Stratos P soluzione 2 o 3",
                        "text": "Testo"
                    }
                }
            },
            "b": {
                "id": 15,
                "type": "question",
                "text": "L'interazione con il pedone deve avvenire tramite pulsante touch (consigliata) o tramite sensore? Scarica le istruzioni posizionamento sensore per valutare se è possibile utilizzarlo nella tua installazione",
                "answers": [
                    {
                        "a": "Sensore"
                    },
                    {
                        "b": "Pulsante touch"
                    }
                ],
                "a": {
                    "id": 16,
                    "type": "question",
                    "text": "Il limite di velocità consentito sulla strada è maggiore di 50km/h?",
                    "answers": [
                        {
                            "a": "no"
                        },
                        {
                            "b": "sì"
                        }
                    ],
                    "a": {
                        "id": 17,
                        "type": "question",
                        "text": "Ci sono ostacoli dove verranno installati i pali che potrebbero ridurre la visibilità dei cartelli di segnalazione laterali (alberi, tabelle segnaletiche o pubblicitarie, ecc)?",
                        "answers": [
                            {
                                "a": "no"
                            },
                            {
                                "b": "sì"
                            }
                        ],
                        "a": {
                            "id": 18,
                            "type": "answer",
                            "title": "Apl smart soluzione 1 con sensore",
                            "text": "Testo"
                        },
                        "b": {
                            "id": 19,
                            "type": "answer",
                            "title": "Apl smart soluzione 2 o 3 con sensore",
                            "text": "Testo"
                        }
                    },
                    "b": {
                        "id": 20,
                        "type": "answer",
                        "title": "Apl smart soluzione 2 o 3 con sensore",
                        "text": "Testo"
                    }
                },
                "b": {
                    "id": 21,
                    "type": "question",
                    "text": "Il limite di velocità consentito sulla strada è maggiore di 50km/h?",
                    "answers": [
                        {
                            "a": "no"
                        },
                        {
                            "b": "sì"
                        }
                    ],
                    "a": {
                        "id": 22,
                        "type": "question",
                        "text": "Ci sono ostacoli dove verranno installati i pali che potrebbero ridurre la visibilità dei cartelli di segnalazione laterali (alberi, tabelle segnaletiche o pubblicitarie, ecc)?",
                        "answers": [
                            {
                                "a": "no"
                            },
                            {
                                "b": "sì"
                            }
                        ],
                        "a": {
                            "id": 23,
                            "type": "answer",
                            "title": "Apl smart soluzione 1 con pulsante touch",
                            "text": "Testo"
                        },
                        "b": {
                            "id": 24,
                            "type": "answer",
                            "title": "Apl smart soluzione 2 o 3 con pulsante touch",
                            "text": "Testo"
                        }
                    },
                    "b": {
                        "id": 25,
                        "type": "answer",
                        "title": "Apl smart soluzione 2 o 3 con pulsante touch",
                        "text": "Testo"
                    }
                }
            }
        }
    }
}