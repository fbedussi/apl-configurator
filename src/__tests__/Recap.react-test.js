import React from 'react';
import Recap from '../components/Recap';
import renderer from 'react-test-renderer';
import texts from './apl-texts.json'

const { questions, labels } = texts;

const breadcrumbs = [{ "0": { "0": { "id": 3, "questionId": null, "answerId": 21 }, "1": { "0": { "id": 5, "questionId": null, "answerId": 16 }, "1": { "id": 6, "questionId": null, "answerId": 1 }, "id": 4, "questionId": 6, "answerId": null }, "id": 2, "questionId": 2, "answerId": null }, "1": { "0": { "0": { "0": { "0": { "0": { "id": 12, "questionId": null, "answerId": 19 }, "1": { "id": 13, "questionId": null, "answerId": 2 }, "id": 11, "questionId": 8, "answerId": null }, "1": { "0": { "id": 15, "questionId": null, "answerId": 20 }, "1": { "id": 16, "questionId": null, "answerId": 23 }, "id": 14, "questionId": 8, "answerId": null }, "id": 10, "questionId": 5, "answerId": null }, "1": { "0": { "0": { "id": 19, "questionId": null, "answerId": 15 }, "1": { "id": 20, "questionId": null, "answerId": 3 }, "id": 18, "questionId": 8, "answerId": null }, "1": { "0": { "id": 22, "questionId": null, "answerId": 14 }, "1": { "id": 23, "questionId": null, "answerId": 5 }, "id": 21, "questionId": 8, "answerId": null }, "id": 17, "questionId": 5, "answerId": null }, "id": 9, "questionId": 4, "images": ["9m.jpg"], "answerId": null }, "1": { "0": { "0": { "id": 26, "questionId": null, "answerId": 15 }, "1": { "id": 27, "questionId": null, "answerId": 3 }, "id": 25, "questionId": 8, "answerId": null }, "1": { "0": { "id": 29, "questionId": null, "answerId": 14 }, "1": { "id": 30, "questionId": null, "answerId": 5 }, "id": 28, "questionId": 8, "answerId": null }, "id": 24, "questionId": 5, "answerId": null }, "id": 8, "questionId": 3, "answerId": null }, "1": { "0": { "0": { "0": { "0": { "id": 35, "questionId": null, "answerId": 6 }, "1": { "id": 36, "questionId": null, "answerId": 13 }, "id": 34, "questionId": 8, "answerid": null }, "1": { "0": { "id": 38, "questionId": null, "answerId": 7 }, "1": { "id": 39, "questionId": null, "answerId": 12 }, "id": 37, "questionId": 8, "answerid": null }, "id": 33, "questionId": 5, "answerId": null }, "1": { "0": { "id": 41, "questionId": null, "answerId": 22 }, "1": { "id": 42, "questionId": null, "answerId": 12 }, "id": 40, "questionId": 8, "answerid": null }, "id": 32, "questionId": 7, "answerId": null }, "1": { "0": { "0": { "0": { "id": 46, "questionId": null, "answerId": 8 }, "1": { "id": 47, "questionId": null, "answerId": 11 }, "id": 45, "questionId": 8, "answerid": null }, "1": { "0": { "id": 49, "questionId": null, "answerId": 9 }, "1": { "id": 50, "questionId": null, "answerId": 10 }, "id": 48, "questionId": 8, "answerid": null }, "id": 44, "questionId": 5, "answerId": null }, "1": { "0": { "id": 52, "questionId": null, "answerId": 9 }, "1": { "id": 53, "questionId": null, "answerId": 10 }, "id": 51, "questionId": 8, "answerid": null }, "id": 43, "questionId": 7, "answerId": null }, "id": 31, "questionId": 6, "answerId": null }, "id": 7, "questionId": 2, "answerId": null }, "id": 1, "questionId": 1, "answerId": null }, { "0": { "0": { "0": { "0": { "0": { "id": 12, "questionId": null, "answerId": 19 }, "1": { "id": 13, "questionId": null, "answerId": 2 }, "id": 11, "questionId": 8, "answerId": null }, "1": { "0": { "id": 15, "questionId": null, "answerId": 20 }, "1": { "id": 16, "questionId": null, "answerId": 23 }, "id": 14, "questionId": 8, "answerId": null }, "id": 10, "questionId": 5, "answerId": null }, "1": { "0": { "0": { "id": 19, "questionId": null, "answerId": 15 }, "1": { "id": 20, "questionId": null, "answerId": 3 }, "id": 18, "questionId": 8, "answerId": null }, "1": { "0": { "id": 22, "questionId": null, "answerId": 14 }, "1": { "id": 23, "questionId": null, "answerId": 5 }, "id": 21, "questionId": 8, "answerId": null }, "id": 17, "questionId": 5, "answerId": null }, "id": 9, "questionId": 4, "images": ["9m.jpg"], "answerId": null }, "1": { "0": { "0": { "id": 26, "questionId": null, "answerId": 15 }, "1": { "id": 27, "questionId": null, "answerId": 3 }, "id": 25, "questionId": 8, "answerId": null }, "1": { "0": { "id": 29, "questionId": null, "answerId": 14 }, "1": { "id": 30, "questionId": null, "answerId": 5 }, "id": 28, "questionId": 8, "answerId": null }, "id": 24, "questionId": 5, "answerId": null }, "id": 8, "questionId": 3, "answerId": null }, "1": { "0": { "0": { "0": { "0": { "id": 35, "questionId": null, "answerId": 6 }, "1": { "id": 36, "questionId": null, "answerId": 13 }, "id": 34, "questionId": 8, "answerid": null }, "1": { "0": { "id": 38, "questionId": null, "answerId": 7 }, "1": { "id": 39, "questionId": null, "answerId": 12 }, "id": 37, "questionId": 8, "answerid": null }, "id": 33, "questionId": 5, "answerId": null }, "1": { "0": { "id": 41, "questionId": null, "answerId": 22 }, "1": { "id": 42, "questionId": null, "answerId": 12 }, "id": 40, "questionId": 8, "answerid": null }, "id": 32, "questionId": 7, "answerId": null }, "1": { "0": { "0": { "0": { "id": 46, "questionId": null, "answerId": 8 }, "1": { "id": 47, "questionId": null, "answerId": 11 }, "id": 45, "questionId": 8, "answerid": null }, "1": { "0": { "id": 49, "questionId": null, "answerId": 9 }, "1": { "id": 50, "questionId": null, "answerId": 10 }, "id": 48, "questionId": 8, "answerid": null }, "id": 44, "questionId": 5, "answerId": null }, "1": { "0": { "id": 52, "questionId": null, "answerId": 9 }, "1": { "id": 53, "questionId": null, "answerId": 10 }, "id": 51, "questionId": 8, "answerid": null }, "id": 43, "questionId": 7, "answerId": null }, "id": 31, "questionId": 6, "answerId": null }, "id": 7, "questionId": 2, "answerId": null }, { "0": { "0": { "0": { "0": { "id": 35, "questionId": null, "answerId": 6 }, "1": { "id": 36, "questionId": null, "answerId": 13 }, "id": 34, "questionId": 8, "answerid": null }, "1": { "0": { "id": 38, "questionId": null, "answerId": 7 }, "1": { "id": 39, "questionId": null, "answerId": 12 }, "id": 37, "questionId": 8, "answerid": null }, "id": 33, "questionId": 5, "answerId": null }, "1": { "0": { "id": 41, "questionId": null, "answerId": 22 }, "1": { "id": 42, "questionId": null, "answerId": 12 }, "id": 40, "questionId": 8, "answerid": null }, "id": 32, "questionId": 7, "answerId": null }, "1": { "0": { "0": { "0": { "id": 46, "questionId": null, "answerId": 8 }, "1": { "id": 47, "questionId": null, "answerId": 11 }, "id": 45, "questionId": 8, "answerid": null }, "1": { "0": { "id": 49, "questionId": null, "answerId": 9 }, "1": { "id": 50, "questionId": null, "answerId": 10 }, "id": 48, "questionId": 8, "answerid": null }, "id": 44, "questionId": 5, "answerId": null }, "1": { "0": { "id": 52, "questionId": null, "answerId": 9 }, "1": { "id": 53, "questionId": null, "answerId": 10 }, "id": 51, "questionId": 8, "answerid": null }, "id": 43, "questionId": 7, "answerId": null }, "id": 31, "questionId": 6, "answerId": null }, { "0": { "0": { "0": { "id": 46, "questionId": null, "answerId": 8 }, "1": { "id": 47, "questionId": null, "answerId": 11 }, "id": 45, "questionId": 8, "answerid": null }, "1": { "0": { "id": 49, "questionId": null, "answerId": 9 }, "1": { "id": 50, "questionId": null, "answerId": 10 }, "id": 48, "questionId": 8, "answerid": null }, "id": 44, "questionId": 5, "answerId": null }, "1": { "0": { "id": 52, "questionId": null, "answerId": 9 }, "1": { "id": 53, "questionId": null, "answerId": 10 }, "id": 51, "questionId": 8, "answerid": null }, "id": 43, "questionId": 7, "answerId": null }, { "0": { "id": 52, "questionId": null, "answerId": 9 }, "1": { "id": 53, "questionId": null, "answerId": 10 }, "id": 51, "questionId": 8, "answerid": null }];
const answersHistory = [1, 1, 1, 1, 1];

test('Recap component', () => {
  const component = renderer.create(
    <Recap
      questions={questions}
      labels={labels}
      breadcrumbs={breadcrumbs}
      answersHistory={answersHistory}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});