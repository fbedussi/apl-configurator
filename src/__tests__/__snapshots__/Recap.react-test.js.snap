// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recap component 1`] = `
<div
  className="recap"
>
  <table
    className="recap-table"
  >
    <thead
      className="recap-head"
    >
      <tr>
        <th>
          domanda
        </th>
        <th>
          risposta
        </th>
      </tr>
    </thead>
    <tbody
      className="recap-body"
    >
      <tr>
        <td
          dangerouslySetInnerHTML={
            Object {
              "__html": "Nel luogo di installazione è possibile collegare il sistema APL alla rete elettrica?",
            }
          }
        />
        <td>
          sì
        </td>
      </tr>
      <tr>
        <td
          dangerouslySetInnerHTML={
            Object {
              "__html": "Il sistema deve essere dotato di dispositivo per l'interazione con il pedone?",
            }
          }
        />
        <td>
          sì
        </td>
      </tr>
      <tr>
        <td
          dangerouslySetInnerHTML={
            Object {
              "__html": "L'interazione con il pedone deve avvenire tramite pulsante touch (consigliata) o tramite sensore? <p class=\\"note\\"><small><em>Scarica le <a href=\\"http://www.attraversamentipedonali.it/pdf/come_installare_sensore_APL_smart.pdf\\" target=\\"_blank\\">istruzioni posizionamento sensore</a> per valutare se è possibile utilizzarlo nella tua installazione</em></small></p>",
            }
          }
        />
        <td>
          Pulsante touch
        </td>
      </tr>
      <tr>
        <td
          dangerouslySetInnerHTML={
            Object {
              "__html": "Il limite di velocità consentito sulla strada è maggiore di 50km/h?",
            }
          }
        />
        <td>
          sì
        </td>
      </tr>
      <tr>
        <td
          dangerouslySetInnerHTML={
            Object {
              "__html": "L'alimentazione 230V è disponibile su entrambi i lati della strada?",
            }
          }
        />
        <td>
          sì
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
